static-assets/components/cstudio-forms/data-sources/img-repository-upload.js

```diff
@@ -21,11 +21,15 @@ function(id, form, properties, constraints)  {
    	this.form = form;
    	this.properties = properties;
    	this.constraints = constraints;
+    this.useSearch = false;
 
-   	for(var i=0; i<properties.length; i++) {
-   		if(properties[i].name == "repoPath") {
- 			this.repoPath = properties[i].value;
-   		}
+    for(let property of properties)
+      if(property.name === "repoPath") {
+        this.repoPath = properties[i].value;
+      }
+      else if(property.name === "useSearch"){
+        this.useSearch = properties[i].value === "true";
+      }
```

Comment:
Looks like `properties` is an array. For clarity, let's just switch the for loop to a `properties.forEach` statement. If it's an object, you may either switch to `Object.entries(properties).forEach` or add a `if (properties.hasOwnProperty(property)` around the for logic.

---

Please format if else with the else inline with the if's curly brace:
```js
} else if (...) {
``` 
instead of 
```js 
} 
else if (...) {
``` 

Same deal with other if/else/else-if, try/catch, etc statements around the PR.